xstac/_kerchunk.py

```diff
@@ -0,0 +1,84 @@
+from typing import Any
+
+import collections
+import json
+import pystac
+
+
+def add_kerchunk_indices(d: dict[str, Any], item: pystac.Item) -> pystac.Item:
+    """
+    Add Kerchunk metadata to a STAC item.
+
+    Notes
+    -----
+    We follow some high-level guidelines:
+
+    1. STAC metadata should always be concrete objects (no JSON strings
+       in the properties)
+    2. We don't duplicate anything already in the STAC metadata
+       (modulo some TODOs around zattrs).
+
+    The output STAC item will have some additional properties:
+
+    - Global `.zgroup` and `.zattrs` are added add `kerchunk:zgroup` and
+      and `kerchunk:zattrs`.
+    - For each dimension and variable, `.zarray` and `.zattrs` are added as
+        `kerchunk:zarray` and `kerchunk:zattrs`.
+    - For each chunk of a variable, the index is added under `kerchunk:value`,
+      as a mapping `kerchunk:value[i.j.k]`.
+    """
+    item = item.clone()
+    refs = d["refs"]
+    for k, v in refs.items():
+        match k.split("/"):
```

Comment:
This requires Python 3.10. I can rewrite it to use an older syntax if needed, but this was the first time I've come across a case that really seemed to benefit from it, so I couldn't help myself :)